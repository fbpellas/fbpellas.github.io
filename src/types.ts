export interface Pronunciation {
  phoneme: string;
  graphemes: string;
  examples: string;
  audioPhoneme: string;
  audioExamples: string;
}

export interface Breadcrumb {
  anchor: string;
  title: string;
}

export enum QuizIndex {
  OddPhonemeOut = 0,
  PhoneticSpelling = 1,
  SameWordsDifferentStress = 2,
  WhereIsTheStress = 3,
  GuessThePattern = 4,
  ShoppingForAPresent = 5
}
