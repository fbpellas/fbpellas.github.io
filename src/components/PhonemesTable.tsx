import * as React from 'react';
import { HiPlay } from 'react-icons/hi';
import { BASE_PATH_SOUNDS } from '../constants';
import { Pronunciation } from '../types';
import { Tooltip } from './Tooltip';

export interface PhonemesTableProps {
  data: Pronunciation[];
}

const PhonemesTable: React.FC<PhonemesTableProps> = (props) => {
  const { data } = props;

  return (
    <table>
      <thead>
        <tr>
          <th>
            <Tooltip className="text-center" definition="Sounds" word="Phonemes" />
          </th>
          <th className="text-center">
            <Tooltip className="text-center" definition="Letters that spell the sound" word="Grapheme" />
          </th>
          <th className="text-center">Examples</th>
        </tr>
      </thead>
      <tbody>
        {data.map((d: Pronunciation) => {
          const { phoneme, graphemes, examples, audioPhoneme, audioExamples } = d;

          const audioPhonemeObj = new Audio(`${BASE_PATH_SOUNDS}${audioPhoneme}`);
          const audioExamplesObj = new Audio(`${BASE_PATH_SOUNDS}${audioExamples}`);
          const examplesSplit = examples.split(', ');

          return (
            <tr key={phoneme}>
              <td>
                <HiPlay
                  className="play-icon"
                  onClick={() => {
                    audioPhonemeObj.play();
                  }}
                />
                {phoneme}
              </td>
              <td>{graphemes}</td>
              <td>
                <HiPlay
                  className="play-icon"
                  onClick={() => {
                    audioExamplesObj.play();
                  }}
                />
                {examplesSplit.map((example: string, index: number) => {
                  const isLast = index === examplesSplit.length - 1;

                  if (isLast)
                    return (
                      <div key={example} className="text-inline nowrap" dangerouslySetInnerHTML={{ __html: example }} />
                    );

                  return (
                    <div key={example} className="text-inline">
                      <div className="text-inline nowrap" dangerouslySetInnerHTML={{ __html: example }} />,{' '}
                    </div>
                  );
                })}
              </td>
            </tr>
          );
        })}
      </tbody>
    </table>
  );
};

export { PhonemesTable };
