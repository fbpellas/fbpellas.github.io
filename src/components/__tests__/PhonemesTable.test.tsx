import React from 'react';
import { render } from '@testing-library/react';
import { PhonemesTable } from '../PhonemesTable';
import { Pronunciation } from '../../types';

test('PhonemesTable', () => {
  const data: Pronunciation[] = [
    {
      phoneme: '/aɪ/',
      graphemes: 'i, igh, y, ie',
      examples: '<u>i</u>ce, b<u>ye</u>, l<u>i</u>me, f<u>igh</u>t, sk<u>y</u>',
      audioPhoneme: 'diphthongs/phonemes/-aɪ-.m4a',
      audioExamples: 'diphthongs/examples/ice, bye, lime, fight , sky.m4a'
    },
    {
      phoneme: '/eɪ/',
      graphemes: 'a, a-e, ai, ay, eigh, ey',
      examples: '<u>a</u>corn, j<u>a</u>d<u>e</u>, p<u>ai</u>d, w<u>eigh</u>t, h<u>ey</u>',
      audioPhoneme: 'diphthongs/phonemes/-eɪ-.m4a',
      audioExamples: 'diphthongs/examples/acorn, jade, paid, weight, hey.m4a'
    }
  ];

  const { container } = render(<PhonemesTable data={data} />);
  expect(container).toMatchSnapshot();
});
