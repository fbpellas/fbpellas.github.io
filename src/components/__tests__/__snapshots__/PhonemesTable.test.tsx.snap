// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhonemesTable 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          <div
            class="text-center"
          >
            Phonemes
          </div>
        </th>
        <th
          class="text-center"
        >
          <div
            class="text-center"
          >
            Grapheme
          </div>
        </th>
        <th
          class="text-center"
        >
          Examples
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <svg
            class="play-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M10 18a8 8 0 100-16 8 8 0 000 16zM9.555 7.168A1 1 0 008 8v4a1 1 0 001.555.832l3-2a1 1 0 000-1.664l-3-2z"
              fill-rule="evenodd"
            />
          </svg>
          /aɪ/
        </td>
        <td>
          i, igh, y, ie
        </td>
        <td>
          <svg
            class="play-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M10 18a8 8 0 100-16 8 8 0 000 16zM9.555 7.168A1 1 0 008 8v4a1 1 0 001.555.832l3-2a1 1 0 000-1.664l-3-2z"
              fill-rule="evenodd"
            />
          </svg>
          <div
            class="text-inline"
          >
            <div
              class="text-inline nowrap"
            >
              <u>
                i
              </u>
              ce
            </div>
            ,
             
          </div>
          <div
            class="text-inline"
          >
            <div
              class="text-inline nowrap"
            >
              b
              <u>
                ye
              </u>
            </div>
            ,
             
          </div>
          <div
            class="text-inline"
          >
            <div
              class="text-inline nowrap"
            >
              l
              <u>
                i
              </u>
              me
            </div>
            ,
             
          </div>
          <div
            class="text-inline"
          >
            <div
              class="text-inline nowrap"
            >
              f
              <u>
                igh
              </u>
              t
            </div>
            ,
             
          </div>
          <div
            class="text-inline nowrap"
          >
            sk
            <u>
              y
            </u>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <svg
            class="play-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M10 18a8 8 0 100-16 8 8 0 000 16zM9.555 7.168A1 1 0 008 8v4a1 1 0 001.555.832l3-2a1 1 0 000-1.664l-3-2z"
              fill-rule="evenodd"
            />
          </svg>
          /eɪ/
        </td>
        <td>
          a, a-e, ai, ay, eigh, ey
        </td>
        <td>
          <svg
            class="play-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M10 18a8 8 0 100-16 8 8 0 000 16zM9.555 7.168A1 1 0 008 8v4a1 1 0 001.555.832l3-2a1 1 0 000-1.664l-3-2z"
              fill-rule="evenodd"
            />
          </svg>
          <div
            class="text-inline"
          >
            <div
              class="text-inline nowrap"
            >
              <u>
                a
              </u>
              corn
            </div>
            ,
             
          </div>
          <div
            class="text-inline"
          >
            <div
              class="text-inline nowrap"
            >
              j
              <u>
                a
              </u>
              d
              <u>
                e
              </u>
            </div>
            ,
             
          </div>
          <div
            class="text-inline"
          >
            <div
              class="text-inline nowrap"
            >
              p
              <u>
                ai
              </u>
              d
            </div>
            ,
             
          </div>
          <div
            class="text-inline"
          >
            <div
              class="text-inline nowrap"
            >
              w
              <u>
                eigh
              </u>
              t
            </div>
            ,
             
          </div>
          <div
            class="text-inline nowrap"
          >
            h
            <u>
              ey
            </u>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
